tss2-dlopen/tss2-dlopen-esys.c

```diff
@@ -0,0 +1,2201 @@
+    if (!sym) {
+        WARN("Function Esys_Initialize not found.");
+        return TSS2_ESYS_RC_NOT_IMPLEMENTED;
+    }
+
+    return sym(esys_context, tcti, abiVersion);
+}
+
+void
+Esys_Finalize(ESYS_CONTEXT **ctx)
+{
+    if (!ctx || !*ctx)
+        return;
+    static TSS2_RC (*sym) (ESYS_CONTEXT **ctx) = NULL;
+    if (!sym)
+        sym = dlsym(esys_dlhandle, "Esys_Finalize");
+    if (!sym) {
+        WARN("Function Esys_Finalize not found.");
+        return;
+    }
+    sym(ctx);
+}
+
+void
+Esys_Free(void *__ptr)
+{
+    if (!__ptr)
+        return;
+    static TSS2_RC (*sym) (void *__ptr) = NULL;
+    if (!sym)
+        sym = dlsym(esys_dlhandle, "Esys_Free");
+    if (!sym) {
+        WARN("Function Esys_Free not found.");
+        return;
+    }
+    sym(__ptr);
+}
+
+#define MAKE_ESYS_0(fun) \
+TSS2_RC fun (ESYS_CONTEXT *ctx) { \
+    static TSS2_RC (*sym) (ESYS_CONTEXT *ctx) = NULL; \
+    if (!sym) \
+        dlsym(esys_dlhandle, str(fun)); \
+    if (!sym) { \
+        WARN("Function " str(fun) " not found."); \
+        return TSS2_ESYS_RC_NOT_IMPLEMENTED; \
+    } \
+    return sym(ctx); \
+}
+
+#define MAKE_ESYS_1(fun,type1,parm1) \
+TSS2_RC fun (ESYS_CONTEXT *ctx, type1 parm1) { \
+    static TSS2_RC (*sym) (ESYS_CONTEXT *ctx, type1) = NULL; \
+    if (!sym) \
+        dlsym(esys_dlhandle, str(fun)); \
+    if (!sym) { \
+        WARN("Function " str(fun) " not found."); \
+        return TSS2_ESYS_RC_NOT_IMPLEMENTED; \
+    } \
+    return sym(ctx, parm1); \
+}
+
+#define MAKE_ESYS_2(fun,type1,parm1,type2,parm2) \
+TSS2_RC fun (ESYS_CONTEXT *ctx, type1 parm1, type2 parm2) { \
+    static TSS2_RC (*sym) (ESYS_CONTEXT *ctx, type1, type2) = NULL; \
+    if (!sym) \
+        dlsym(esys_dlhandle, str(fun)); \
+    if (!sym) { \
+        WARN("Function " str(fun) " not found."); \
+        return TSS2_ESYS_RC_NOT_IMPLEMENTED; \
+    } \
+    return sym(ctx, parm1, parm2); \
+}
+
+#define MAKE_ESYS_3(fun,type1,parm1,type2,parm2,type3,parm3) \
+TSS2_RC fun (ESYS_CONTEXT *ctx, type1 parm1, type2 parm2, type3 parm3) { \
+    static TSS2_RC (*sym) (ESYS_CONTEXT *ctx, type1, type2, type3) = NULL; \
+    if (!sym) \
+        dlsym(esys_dlhandle, str(fun)); \
+    if (!sym) { \
+        WARN("Function " str(fun) " not found."); \
+        return TSS2_ESYS_RC_NOT_IMPLEMENTED; \
+    } \
+    return sym(ctx, parm1, parm2, parm3); \
+}
+
+#define MAKE_ESYS_4(fun,type1,parm1,type2,parm2,type3,parm3,type4,parm4) \
+TSS2_RC fun (ESYS_CONTEXT *ctx, type1 parm1, type2 parm2, type3 parm3, type4 parm4) { \
+    static TSS2_RC (*sym) (ESYS_CONTEXT *ctx, type1, type2, type3, type4) = NULL; \
+    if (!sym) \
+        dlsym(esys_dlhandle, str(fun)); \
+    if (!sym) { \
+        WARN("Function " str(fun) " not found."); \
+        return TSS2_ESYS_RC_NOT_IMPLEMENTED; \
+    } \
+    return sym(ctx, parm1, parm2, parm3, parm4); \
+}
+
+#define MAKE_ESYS_5(fun, type1,parm1, type2,parm2, type3,parm3, type4,parm4, \
```

Comment:
for the other macros e.g. MAKE_ESYS_4 you don't "group" type,param but skip the spaces alltogether.
please make this consisten.

Reply:
resolved but not changed?